.github/workflows/holochain-build-and-test.yml

```diff
@@ -84,3 +76,52 @@ jobs:
           #   export basecmd="nix run -L --show-trace .#packages.${system}."
           #   ${basecmd}${{ join(matrix.cmd.pkgs, '; ${basecmd}') }}
           # fi
+
+  # ensures the cache is regularly updated for the supported versions on multiple platforms
+  cache-update:
+    # This only runs on develop because it points at develop anyway, no need for maintenance branches to do the same job
+    # if: github.ref == 'refs/heads/develop'
+    strategy:
+      fail-fast: false
+      matrix:
+        cmd:
+          - extra_arg: "--override-input versions \"github:holochain/holochain?dir=versions/0_2\""
+          - extra_arg: "--override-input versions \"github:holochain/holochain?dir=versions/0_1\""
+          - extra_arg: "--override-input versions \"github:holochain/holochain?dir=versions/weekly\""
+        platform:
+          - system: x86_64-darwin
+          - system: aarch64-darwin
+          - system: x86_64-linux
+
+    runs-on: [ self-hosted, multi-arch ]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          lfs: false
+
+      - name: Print matrix
+        env:
+          MATRIX: ${{ toJSON(matrix) }}
+        run: echo ${MATRIX}
+
+      - name: "Cache packages ${{ join(matrix.cmd.pkgs, ',') }}"
+        env:
+          system: ${{ matrix.platform.system }}
+          NIX_CONFIG: "access-tokens = github.com=${{ secrets.GITHUB_TOKEN }}"
+          CACHIX_AUTH_TOKEN: "${{ secrets.TS_CACHIX_TOKEN }}"
+        run: |
+          set -xe
+
+          # if a nix store path was passed, ensure it exists locally
+          if [[ ${{ inputs.repo_path }} == "/nix/"* ]]; then
+            nix-store --realise ${{inputs.repo_path }}
+          fi
+
+          # See https://docs.cachix.org/pushing#id1 for pushing shell environments
+          profile_name=holonix-cache-${system}
+          nix develop --profile $profile_name -c true \
```

Comment:
1. i like this idea. i'm not sure if all intermediary build dependencies will end up in the profile, that's something to double-check.
2. `nix develop` doesn't work for remote builders, only `nix build` commands will.
3. also, if using `nix build` it'll still be pushed by the builder's system cachix-watch daemon so we'd have to separate all builders into profiles and push explicitly.